sycl/include/sycl/ext/intel/esimd/detail/memory_intrin.hpp

```diff
@@ -1164,10 +1164,6 @@ __ESIMD_INTRIN void __esimd_media_st(TACC handle, unsigned x, unsigned y,
 }
 #endif // __SYCL_DEVICE_ONLY__
 
-// getter methods returning surface index are not available when stateless
-// memory accesses are enforced.
-#ifndef __ESIMD_FORCE_STATELESS_MEM
-
```

Comment:
this change seems a bit weird to me, what is a case where we want sureface index in stateless mode?

Reply:
'media' API working with image accessors doesn't have any alternative like buffer-accessors (svmptr_t vs buffer_t).
Image-accessors are lowered the same way with and without force-stateless switch: https://github.com/intel/llvm/blob/sycl/llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp#L1656

I don't see why not let users control the usage of media API. Let's say if the source has 2 kernels: one kernel with image and another with buffer_t. It should be possible to compile such source with split-per-kernel setting and run them (or not run) depending on if image is supported on target.